src/callbacks.rs

```diff
@@ -76,4 +78,21 @@ pub trait ParseCallbacks: fmt::Debug + UnwindSafe {
 
     /// This will be called on every file inclusion, with the full path of the included file.
     fn include_file(&self, _filename: &str) {}
+
+    /// This will be called to determine whether a particular blocklisted type
+    /// implements a trait or not. This will be used to implement traits on
+    /// other types containing the blocklisted type.
+    ///
+    /// * `None`: use the default behavior
+    /// * `Some(ImplementsTrait::Yes)`: `_name` implements `_derive_trait`
+    /// * `Some(ImplementsTrait::Manually)`: any type including `_name` can't
+    ///   derive `_derive_trait` but can implemented it manually
+    /// * `Some(ImplementsTrait::No)`: `_name` doesn't implement `_derive_trait`
+    fn blocklisted_type_implements_trait(
+        &self,
+        _name: &str,
+        _derive_trait: DeriveTrait,
+    ) -> Option<ImplementsTrait> {
```

Comment:
Is there any reason this can't just return a `bool`? I think we probably never want the default behavior to be other than `No`, tbf, and the `manually` behavior seems relatively hard to understand, so I'd prefer if the API was simpler (unless you think there's a good use case for that?)

Reply:
> the manually behavior seems relatively hard to understand

Yes, this is #2005

> unless you think there's a good use case for that?

I do have a usecase but not sure it qualifies as “good”. bindgen never thinks `pthread_mutex_t` should implement Default. 0.19 used to generate `mem::zeroed` defaults for structs containing `pthread_mutex_t`. With this check, I end up getting that behavior:
```
        if name == "pthread_mutex_t" && derive_trait == bindgen::callbacks::DeriveTrait::Default {
            Some(bindgen::callbacks::ImplementsTrait::Manually)
        }
```

Reply:
I guess I'm not super-excited about it, but your call. If you need it I'm ok merging as-is.

Reply:
Great, thanks. We can revisit this if/when #2005 gets addressed.